app/backend/src/couchers/servicers/pages.py

```diff
@@ -4,16 +4,17 @@
 
 from couchers import errors
 from couchers.db import session_scope
-from couchers.models import Page, PageType, PageVersion, User
+from couchers.models import Cluster, Node, Page, PageType, PageVersion, User
 from couchers.utils import Timestamp_from_datetime, create_coordinate, remove_duplicates_retain_order, slugify
 from pb import pages_pb2, pages_pb2_grpc
 
 
 def _check_update_permission(page: Page, user_id):
-    if page.owner_user_id == user_id:
-        return True
-    # elif page.owner_cluster_id # TODO
-    return False
+    if page.owner_user:
+        return page.owner_user_id == user_id
+    elif page.owner_cluster:
+        return page.owner_cluster.admins.filter(User.id == user_id).one_or_none() is not None
+    raise Exception("Page must have an owner.")
```

Comment:
raise a `RuntimeError` if it is a programming mistake (preferably with an assert statement).

Can we add a database invariant that ensures a page has a owner?

Reply:
There should be a database invariant. Maybe I can just remove the raise?

Reply:
Yes and the elif line. Then we will get a `NoneType has no attribute admins` runtime error which is as good as an assert.